language/move-lang/src/parser/syntax.rs

```diff
@@ -223,16 +223,41 @@ fn parse_identifier(tokens: &mut Lexer) -> Result<Name, Error> {
     Ok(spanned(tokens.file_name(), start_loc, end_loc, id))
 }
 
-// Parse an account address:
-//      Address = <Number>
-fn parse_address(tokens: &mut Lexer) -> Result<Address, Error> {
-    if tokens.peek() != Tok::NumValue {
-        return Err(unexpected_token_error(tokens, "an account address value"));
+// Parse a numerical address value
+//     AddressBytes = <Number>
+fn parse_address_bytes(tokens: &mut Lexer) -> Result<Spanned<AddressBytes>, Error> {
+    let loc = current_token_loc(tokens);
+    let addr_res = AddressBytes::parse_str(tokens.content());
+    consume_token(tokens, Tok::NumValue)?;
+    match addr_res {
+        Ok(addr_) => Ok(sp(loc, addr_)),
+        Err(msg) => Err(vec![(loc, msg)]),
+    }
+}
+
+// Helper for parse_leading_name_access_with_description with default description
+fn parse_leading_name_access(tokens: &mut Lexer) -> Result<LeadingNameAccess, Error> {
+    parse_leading_name_access_with_description(tokens, || "an address or an identifier")
+}
+
+// Parse an the beginning of an acces, either an address or an identifier:
+//      LeadingNameAccess = <AddressBytes> | <Identifier>
+fn parse_leading_name_access_with_description<'a, F: FnOnce() -> &'a str>(
```

Comment:
In the case of `parse_name_access_chain`, you used an underscore suffix instead of `_with_description`, and you put the grammar comment on the function without the description argument. I kind of like that better. At least they should be consistent